pom.xml

```diff
@@ -120,6 +120,19 @@
         <artifactId>logback-classic</artifactId>
         <version>1.2.2</version>
       </dependency>
+
+
+      <!-- Bean validation -->
+      <dependency>
+        <groupId>org.hibernate</groupId>
+        <artifactId>hibernate-validator</artifactId>
```

Comment:
This is the reference implementation for JSR-303 Bean Validation offered by hiberante project. Not to confuse with any of their JPA persistence stuff.
However you can integrate it into JPA. If you use JPA your Bean Validation is automatically checked on PrePersist and PreUpdate as far as I know.